C3real-kill3r/H-digest
# Ft enable signup and login 167264802

**What does this PR do?**
 Includes the implementation of user signup and user login
- [x] Enable user signup
    This is done by creating a user Viewset that allows for user creation
- [x] Enable user login
  This is done by incorporating the package  `restframework_simplejwt` which allows 
you to generatre jwt tokens that will be used for authentication
- [x] Set up the groups for role based authentication
 I also took the chance to setup the groups viewsets which will allow for the implementation of role based authorization


**Any background context you want to provide or addition**
N/A

**How to test**
To test this you should
- clone this repo `git clone https://github.com/C3real-kill3r/NCPWD.git`
- checkout to this branch `git checkout ft-enable-signup-and-login-167264802`
- run the server `./manage.py runserver`
- visit `localhost::8000/api/users`  to see the varius options for users including user creation
- visit `localhost::8000/api/token` to be allowed to login and get a token

**What are the relevant pivotal tracker stories?**
[#167264802](https://www.pivotaltracker.com/story/show/167264802)
[#167264823](https://www.pivotaltracker.com/story/show/167264823)


**Questions:**
